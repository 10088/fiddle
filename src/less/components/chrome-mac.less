.darwin {
  .chrome {
    text-align: center;
    background-color: @background-1;
    border-bottom: @border;
    -webkit-app-region: drag;
    height: 23px;
    font-size: 1.4rem;

    small,
    p,
    span {
      user-select: none;
    }
  }
}
