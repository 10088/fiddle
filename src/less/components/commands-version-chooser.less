button.version-chooser.bp3-button {
  border-radius: 3px;
}
