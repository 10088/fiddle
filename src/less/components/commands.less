@import "../variables.less";

header {
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.2), 0 0 0 rgba(16, 22, 26, 0), 0 1px 1px rgba(16, 22, 26, 0.4);
  background-color: @background-1;
  -webkit-app-region: drag;
}

.commands {
  display: flex;
  justify-content: space-between;
  height: 50px;
  padding: 10px;
  box-sizing: border-box;
  font-family: @fonts-common;

  button,
  input,
  select {
    -webkit-app-region: no-drag;
    margin-bottom: 0;
  }

  select {
    width: 100px;
    margin-left: 5px;
  }

  & > div {
    display: flex;

    &:last-of-type {
      margin-right: 10px;

      button {
        margin-left: 5px;
      }
    }
  }

  .address-bar {
    height: 28px;
    width: 360px;
    transition: width 0.4s;
    display: flex;

    input,
    span {
      font-family: @fonts-common;
      font-weight: 300;
      font-size: 11px;
      cursor: auto;
      -webkit-app-region: no-drag;
    }

    input {
      flex-grow: 1;
      background: white;
      padding-left: 1rem;
      height: 2.6rem;
    }

    input:invalid {
      box-shadow: inset 0 0 0 1px #e4252d;
    }

    span {
      background: white;
      color: #1d1d1d;
      border-bottom-color: rgb(209, 209, 209);
      border-bottom-style: solid;
      border-bottom-left-radius: 4px;
      border-bottom-width: 1px;
      border-image-outset: 0;
      border-image-repeat: stretch;
      border-image-slice: 100%;
      border-image-source: none;
      border-image-width: 1;
      border-left-color: rgb(209, 209, 209);
      border-left-style: solid;
      border-left-width: 1px;
      border-top-color: rgb(209, 209, 209);
      border-top-left-radius: 4px;
      border-top-style: solid;
      border-top-width: 1px;
      display: inline-block;
      height: 1.2rem;
      letter-spacing: normal;
      margin: 0;
      padding: 6px 0 10px 10px;
      user-select: text;
      width: 132px;
      text-rendering: auto;
      text-shadow: none;
      text-transform: none;
    }

    &.empty {
      width: 200px;
    }

    &:focus-within {
      span,
      input {
        border-color: @foreground-2;
      }
    }
  }
}
