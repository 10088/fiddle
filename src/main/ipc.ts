import { EventEmitter } from 'events';
import { ipcMain } from 'electron';
import { ipcMainEvents } from '../ipc-events';

/**
 * The main purpose of this class is to be the central
 * gathering place for IPC calls the main process sends
 * or listens to.
 *
 * @class IpcManager
 * @extends {EventEmitter}
 */
export class IpcMainManager extends EventEmitter {
  constructor() {
    super();

    ipcMainEvents.forEach((name) => {
      ipcMain.on(name, (...args) => this.emit(name, ...args));
    });
  }
}

export const ipcMainManager = new IpcMainManager();
