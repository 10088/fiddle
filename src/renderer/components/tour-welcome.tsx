import * as React from 'react';
import { observer } from 'mobx-react';

import { AppState } from '../state';
import { TourScriptStep, Tour } from './tour';
import { Dialog } from './dialog';

export interface WelcomeTourProps {
  appState: AppState;
}

export interface WelcomeTourState {
  isTourStarted: boolean;
}

const welcomeTour: Set<TourScriptStep> = new Set([
  {
    selector: 'div.mosaic-window.main',
    text: 'Main editor'
  }
]);

@observer
export class WelcomeTour extends React.Component<WelcomeTourProps, WelcomeTourState> {
  constructor(props: WelcomeTourProps) {
    super(props);

    this.stopTour = this.stopTour.bind(this);
    this.startTour = this.startTour.bind(this);

    this.state = {
      isTourStarted: false
    };
  }

  public stopTour() {
    this.props.appState.isTourShowing = false;
  }

  public startTour() {
    this.setState({ isTourStarted: true });
  }

  public render() {
    const { isTourShowing } = this.props.appState;
    const { isTourStarted } = this.state;

    if (!isTourShowing) return null;

    if (!isTourStarted) {
      return (
        <Dialog
          key='welcome-tour-dialog'
          isCentered={true}
          isShowing={true}
          isShowingBackdrop={true}
          onConfirm={this.startTour}
          onClose={this.stopTour}
        >
          <span>Would you like to start the tour?</span>
        </Dialog>
      );
    } else {
      return (
        <Tour tour={welcomeTour} />
      );
    }
  }
}
