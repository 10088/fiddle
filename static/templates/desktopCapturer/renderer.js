const { desktopCapturer } = require('electron')

// The following example shows how to capture video from
// the screen. It also grabs each window, so you could
// just grab video from a single window.
//
desktopCapturer.getSources({ types: ['window', 'screen'] }, (error, sources) => {
  if (error) throw error
  for (let i = 0; i < sources.length; ++i) {
    console.log(sources[i])

    if (sources[i].name.startsWith('Screen')) {
      navigator.mediaDevices.getUserMedia({
        audio: false,
        video: {
          mandatory: {
            chromeMediaSource: 'desktop',
            chromeMediaSourceId: sources[i].id,
            minWidth: 1280,
            maxWidth: 1280,
            minHeight: 720,
            maxHeight: 720
          }
        }
      })
      .then((stream) => handleStream(stream))
      .catch((e) => handleError(e))
      return
    }
  }
})

function handleStream (stream) {
  const video = document.querySelector('video')
  video.srcObject = stream
  video.onloadedmetadata = (e) => video.play()
}

function handleError (e) {
  console.log(e)
}