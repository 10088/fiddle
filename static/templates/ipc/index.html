<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>IPC Example</title>
</head>
<body>
  <h1>IPC Example</h1>
  <p>Open the developer tools to see the IPC in action.</p>
</body>
</html>
