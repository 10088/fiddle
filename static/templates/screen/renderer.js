const { screen } = require('electron')

const screens = screen.getAllDisplays()
const div = document.querySelector('div')

div.innerText = JSON.stringify(screens);
