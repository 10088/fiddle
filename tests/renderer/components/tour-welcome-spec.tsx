import { shallow } from 'enzyme';
import * as React from 'react';

import { WelcomeTour } from '../../../src/renderer/components/tour-welcome';
import { ipcRendererManager } from '../../../src/renderer/ipc';

describe('Header component', () => {
  beforeEach(() => {
    this.store = {
      isTourShowing: true,
      disableTour: jest.fn()
    };

    ipcRendererManager.removeAllListeners();
  });

  it('renders', () => {
    const wrapper = shallow(<WelcomeTour appState={this.store} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the tour once started', () => {
    const wrapper = shallow(<WelcomeTour appState={this.store} />);
    const instance: WelcomeTour = wrapper.instance() as any;

    instance.startTour();

    expect(wrapper.state('isTourStarted')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('stops the tour on stopTour()', () => {
    const wrapper = shallow(<WelcomeTour appState={this.store} />);
    const instance: WelcomeTour = wrapper.instance() as any;

    instance.stopTour();

    expect(wrapper.state('isTourStarted')).toBe(false);
    expect(this.store.disableTour).toHaveBeenCalled();
  });
});
