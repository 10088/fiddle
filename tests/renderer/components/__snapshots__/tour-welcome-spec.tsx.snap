// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders 1`] = `
<Dialog
  isCentered={true}
  isShowing={true}
  isShowingBackdrop={true}
  key="welcome-tour-dialog"
  onClose={[Function]}
  onConfirm={[Function]}
>
  <h4>
    🙋‍ Hey There!
  </h4>
  <p>
    Welcome to Electron Fiddle! If you're new to the app, we'd like to give you a brief tour of its features.
  </p>
  <p>
    We won't show this dialog again, but you can always find the tour in the Help menu.
  </p>
</Dialog>
`;

exports[`Header component renders the tour once started 1`] = `
<Dialog
  isCentered={true}
  isShowing={true}
  isShowingBackdrop={true}
  key="welcome-tour-dialog"
  onClose={[Function]}
  onConfirm={[Function]}
>
  <h4>
    🙋‍ Hey There!
  </h4>
  <p>
    Welcome to Electron Fiddle! If you're new to the app, we'd like to give you a brief tour of its features.
  </p>
  <p>
    We won't show this dialog again, but you can always find the tour in the Help menu.
  </p>
</Dialog>
`;
